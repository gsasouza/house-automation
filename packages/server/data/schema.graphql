# This file was generated based on ".graphqlconfig". Do not edit manually.

schema {
  query: Query
  mutation: Mutation
  subscription: Subscription
}

"An object with an ID"
interface Node {
  "The id of the object."
  id: ID!
}

union Me = AdminUser | User

"Admin User data"
type AdminUser implements Node {
  _id: String
  email: String
  "The ID of an object"
  id: ID!
  name: String
  username: String
}

type AdminUserLoginPayload {
  clientMutationId: String
  error: String
  token: String
}

"Board data"
type Board implements Node {
  _id: String
  connected: Boolean
  createdBy: User
  host: String
  "The ID of an object"
  id: ID!
  name: String
  port: String
  type: String
}

type BoardAddPayload {
  boardEdge: BoardEdge
  clientMutationId: String
  error: String
}

"A connection to a list of items."
type BoardConnection {
  count: Int
  "A list of edges."
  edges: [BoardEdge]
  "Information to aid in pagination."
  pageInfo: PageInfo!
}

"An edge in a connection."
type BoardEdge {
  "A cursor for use in pagination"
  cursor: String!
  "The item at the end of the edge"
  node: Board
}

"Device data"
type Device implements Node {
  _id: String
  board: Board
  connected: Boolean
  createdBy: User
  "The ID of an object"
  id: ID!
  name: String
  pin: String
  room: Device
  state: Boolean
  type: String
}

type DeviceAddPayload {
  clientMutationId: String
  deviceEdge: DeviceEdge
  error: String
}

type DeviceChangeStatePayload {
  clientMutationId: String
  deviceEdge: DeviceEdge
  error: String
}

type DeviceChangedStatePayload {
  deviceEdge: DeviceEdge
}

"A connection to a list of items."
type DeviceConnection {
  count: Int
  "A list of edges."
  edges: [DeviceEdge]
  "Information to aid in pagination."
  pageInfo: PageInfo!
}

"An edge in a connection."
type DeviceEdge {
  "A cursor for use in pagination"
  cursor: String!
  "The item at the end of the edge"
  node: Device
}

type Mutation {
  AdminUserLogin(input: AdminUserLoginInput!): AdminUserLoginPayload
  BoardAdd(input: BoardAddInput!): BoardAddPayload
  DeviceAdd(input: DeviceAddInput!): DeviceAddPayload
  DeviceChangeState(input: DeviceChangeStateInput!): DeviceChangeStatePayload
  PlaceAdd(input: PlaceAddInput!): PlaceAddPayload
  RoomAdd(input: RoomAddInput!): RoomAddPayload
  UserLogin(input: UserLoginInput!): UserLoginPayload
}

"Information about pagination in a connection."
type PageInfo {
  "When paginating forwards, the cursor to continue."
  endCursor: String
  "When paginating forwards, are there more items?"
  hasNextPage: Boolean!
  "When paginating backwards, are there more items?"
  hasPreviousPage: Boolean!
  "When paginating backwards, the cursor to continue."
  startCursor: String
}

"Place data"
type Place implements Node {
  _id: String
  "The ID of an object"
  id: ID!
  name: String
  reference: String
}

type PlaceAddPayload {
  clientMutationId: String
  error: String
  placeEdge: PlaceEdge
}

"A connection to a list of items."
type PlaceConnection {
  count: Int
  "A list of edges."
  edges: [PlaceEdge]
  "Information to aid in pagination."
  pageInfo: PageInfo!
}

"An edge in a connection."
type PlaceEdge {
  "A cursor for use in pagination"
  cursor: String!
  "The item at the end of the edge"
  node: Place
}

"The root of all... queries"
type Query {
  boards(after: String, before: String, first: Int, last: Int, search: String): BoardConnection
  devices(after: String, before: String, first: Int, last: Int, search: String): DeviceConnection
  me: Me
  "Fetches an object given its ID"
  node(
    #The ID of an object
    id: ID!
  ): Node
  places(after: String, before: String, first: Int, last: Int, search: String): PlaceConnection
  rooms(after: String, before: String, first: Int, last: Int, search: String): RoomConnection
  users(after: String, before: String, first: Int, last: Int, search: String): UserConnection
}

"Room data"
type Room implements Node {
  _id: String
  boardIosConnected(after: String, before: String, first: Int, last: Int, search: String): DeviceConnection
  boardIosConnectedCount: Int
  createdBy: User
  "The ID of an object"
  id: ID!
  name: String
  type: String
}

type RoomAddPayload {
  clientMutationId: String
  error: String
  roomEdge: RoomEdge
}

"A connection to a list of items."
type RoomConnection {
  count: Int
  "A list of edges."
  edges: [RoomEdge]
  "Information to aid in pagination."
  pageInfo: PageInfo!
}

"An edge in a connection."
type RoomEdge {
  "A cursor for use in pagination"
  cursor: String!
  "The item at the end of the edge"
  node: Room
}

type Subscription {
  DeviceChangedState: DeviceChangedStatePayload
}

"User data"
type User implements Node {
  _id: String
  "The ID of an object"
  id: ID!
  isAdmin: Boolean
  name: String
  username: String
}

"A connection to a list of items."
type UserConnection {
  count: Int
  "A list of edges."
  edges: [UserEdge]
  "Information to aid in pagination."
  pageInfo: PageInfo!
}

"An edge in a connection."
type UserEdge {
  "A cursor for use in pagination"
  cursor: String!
  "The item at the end of the edge"
  node: User
}

type UserLoginPayload {
  clientMutationId: String
  error: String
  token: String
}

input AdminUserLoginInput {
  clientMutationId: String
  password: String!
  username: String!
}

input BoardAddInput {
  clientMutationId: String
  host: String
  name: String!
  port: String
  type: String!
}

input DeviceAddInput {
  board: ID!
  clientMutationId: String
  name: String!
  pin: Int!
  room: ID!
  type: String!
}

input DeviceChangeStateInput {
  clientMutationId: String
  id: ID!
  state: Boolean!
}

input PlaceAddInput {
  clientMutationId: String
  name: String!
  reference: String!
}

input RoomAddInput {
  clientMutationId: String
  name: String!
  type: String!
}

input UserLoginInput {
  clientMutationId: String
  password: String!
  username: String!
}
