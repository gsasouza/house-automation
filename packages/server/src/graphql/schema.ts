import MutationType from './type/MutationType';
import QueryType from './type/QueryType';
import SubscriptionType from './type/SubscriptionType';

import { GraphQLSchema } from 'graphql';

const schema = new GraphQLSchema({
  query: QueryType,
  mutation: MutationType,
  subscription: SubscriptionType,
});

export default schema;
