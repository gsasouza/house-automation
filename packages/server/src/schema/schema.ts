import { GraphQLSchema } from 'graphql';

import QueryType from './QueryType';
import MutationType from './MutationType';

export default new GraphQLSchema({
  query: QueryType,
  mutation: MutationType,
});
