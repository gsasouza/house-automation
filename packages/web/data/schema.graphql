input AddRoomInput {
  name: String!
  type: String!
  clientMutationId: String
}

type AddRoomPayload {
  roomEdge: RoomEdge
  error: String
  clientMutationId: String
}

input LoginInput {
  username: String!
  password: String!
  clientMutationId: String
}

type LoginPayload {
  token: String
  error: String
  clientMutationId: String
}

type Mutation {
  Login(input: LoginInput!): LoginPayload
  AddRoom(input: AddRoomInput!): AddRoomPayload
}

"""An object with an ID"""
interface Node {
  """The id of the object."""
  id: ID!
}

"""Information about pagination in a connection."""
type PageInfo {
  """When paginating forwards, are there more items?"""
  hasNextPage: Boolean!

  """When paginating backwards, are there more items?"""
  hasPreviousPage: Boolean!

  """When paginating backwards, the cursor to continue."""
  startCursor: String

  """When paginating forwards, the cursor to continue."""
  endCursor: String
}

"""The root of all... queries"""
type Query {
  """Fetches an object given its ID"""
  node(
    """The ID of an object"""
    id: ID!
  ): Node
  users(after: String, first: Int, before: String, last: Int, search: String): UserConnection
  rooms(after: String, first: Int, before: String, last: Int, search: String): RoomConnection
}

"""Room data"""
type Room implements Node {
  """The ID of an object"""
  id: ID!
  _id: String
  name: String
  type: String
  createdBy: User
}

"""A connection to a list of items."""
type RoomConnection {
  """Information to aid in pagination."""
  pageInfo: PageInfo!

  """A list of edges."""
  edges: [RoomEdge]
  count: Int
}

"""An edge in a connection."""
type RoomEdge {
  """The item at the end of the edge"""
  node: Room

  """A cursor for use in pagination"""
  cursor: String!
}

"""User data"""
type User implements Node {
  """The ID of an object"""
  id: ID!
  _id: String
  name: String
  isAdmin: Boolean
  username: String
}

"""A connection to a list of items."""
type UserConnection {
  """Information to aid in pagination."""
  pageInfo: PageInfo!

  """A list of edges."""
  edges: [UserEdge]
  count: Int
}

"""An edge in a connection."""
type UserEdge {
  """The item at the end of the edge"""
  node: User

  """A cursor for use in pagination"""
  cursor: String!
}
