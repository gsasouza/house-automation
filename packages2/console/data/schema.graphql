"""Admin User data"""
type AdminUser implements Node {
  """The ID of an object"""
  id: ID!
  _id: String
  name: String
  username: String
  email: String
}

input AdminUserLoginInput {
  username: String!
  password: String!
  clientMutationId: String
}

type AdminUserLoginPayload {
  token: String
  error: String
  clientMutationId: String
}

"""Board data"""
type Board implements Node {
  """The ID of an object"""
  id: ID!
  _id: String
  name: String
  type: String
  host: String
  port: String
  connected: Boolean
  createdBy: User
}

input BoardAddInput {
  name: String!
  type: String!
  port: String
  host: String
  clientMutationId: String
}

type BoardAddPayload {
  boardEdge: BoardEdge
  error: String
  clientMutationId: String
}

"""A connection to a list of items."""
type BoardConnection {
  """Information to aid in pagination."""
  pageInfo: PageInfo!

  """A list of edges."""
  edges: [BoardEdge]
  count: Int
}

"""An edge in a connection."""
type BoardEdge {
  """The item at the end of the edge"""
  node: Board

  """A cursor for use in pagination"""
  cursor: String!
}

"""Device data"""
type Device implements Node {
  """The ID of an object"""
  id: ID!
  _id: String
  name: String
  type: String
  board: Board
  pin: String
  state: Boolean
  connected: Boolean
  room: Device
  createdBy: User
}

input DeviceAddInput {
  name: String!
  type: String!
  pin: Int!
  board: ID!
  room: ID!
  clientMutationId: String
}

type DeviceAddPayload {
  deviceEdge: DeviceEdge
  error: String
  clientMutationId: String
}

type DeviceChangedStatePayload {
  deviceEdge: DeviceEdge
}

input DeviceChangeStateInput {
  id: ID!
  state: Boolean!
  clientMutationId: String
}

type DeviceChangeStatePayload {
  deviceEdge: DeviceEdge
  error: String
  clientMutationId: String
}

"""A connection to a list of items."""
type DeviceConnection {
  """Information to aid in pagination."""
  pageInfo: PageInfo!

  """A list of edges."""
  edges: [DeviceEdge]
  count: Int
}

"""An edge in a connection."""
type DeviceEdge {
  """The item at the end of the edge"""
  node: Device

  """A cursor for use in pagination"""
  cursor: String!
}

union Me = User | AdminUser

type Mutation {
  UserLogin(input: UserLoginInput!): UserLoginPayload
  RoomAdd(input: RoomAddInput!): RoomAddPayload
  BoardAdd(input: BoardAddInput!): BoardAddPayload
  DeviceAdd(input: DeviceAddInput!): DeviceAddPayload
  DeviceChangeState(input: DeviceChangeStateInput!): DeviceChangeStatePayload
  AdminUserLogin(input: AdminUserLoginInput!): AdminUserLoginPayload
}

"""An object with an ID"""
interface Node {
  """The id of the object."""
  id: ID!
}

"""Information about pagination in a connection."""
type PageInfo {
  """When paginating forwards, are there more items?"""
  hasNextPage: Boolean!

  """When paginating backwards, are there more items?"""
  hasPreviousPage: Boolean!

  """When paginating backwards, the cursor to continue."""
  startCursor: String

  """When paginating forwards, the cursor to continue."""
  endCursor: String
}

"""The root of all... queries"""
type Query {
  """Fetches an object given its ID"""
  node(
    """The ID of an object"""
    id: ID!
  ): Node
  me: Me
  users(after: String, first: Int, before: String, last: Int, search: String): UserConnection
  rooms(after: String, first: Int, before: String, last: Int, search: String): RoomConnection
  boards(after: String, first: Int, before: String, last: Int, search: String): BoardConnection
  devices(after: String, first: Int, before: String, last: Int, search: String): DeviceConnection
}

"""Room data"""
type Room implements Node {
  """The ID of an object"""
  id: ID!
  _id: String
  name: String
  type: String
  createdBy: User
  boardIosConnectedCount: Int
  boardIosConnected(after: String, first: Int, before: String, last: Int, search: String): DeviceConnection
}

input RoomAddInput {
  name: String!
  type: String!
  clientMutationId: String
}

type RoomAddPayload {
  roomEdge: RoomEdge
  error: String
  clientMutationId: String
}

"""A connection to a list of items."""
type RoomConnection {
  """Information to aid in pagination."""
  pageInfo: PageInfo!

  """A list of edges."""
  edges: [RoomEdge]
  count: Int
}

"""An edge in a connection."""
type RoomEdge {
  """The item at the end of the edge"""
  node: Room

  """A cursor for use in pagination"""
  cursor: String!
}

type Subscription {
  DeviceChangedState: DeviceChangedStatePayload
}

"""User data"""
type User implements Node {
  """The ID of an object"""
  id: ID!
  _id: String
  name: String
  isAdmin: Boolean
  username: String
}

"""A connection to a list of items."""
type UserConnection {
  """Information to aid in pagination."""
  pageInfo: PageInfo!

  """A list of edges."""
  edges: [UserEdge]
  count: Int
}

"""An edge in a connection."""
type UserEdge {
  """The item at the end of the edge"""
  node: User

  """A cursor for use in pagination"""
  cursor: String!
}

input UserLoginInput {
  username: String!
  password: String!
  clientMutationId: String
}

type UserLoginPayload {
  token: String
  error: String
  clientMutationId: String
}
