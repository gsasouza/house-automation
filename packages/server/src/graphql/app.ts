import { subscriptionsSetup } from '@housejs/pubsub';

import { authenticatedMiddleware } from '../common/auth';
import { GRAPHQL_PORT, PUBNUB_CREDENTIALS } from '../common/config';
import { dataloadersMiddleware } from './loaders';
import schema from './schema';

import koaPlayground from 'graphql-playground-middleware-koa';
import cors from 'kcors';
import Koa from 'koa';
import graphqlHTTP from 'koa-graphql';
import morgan from 'koa-morgan';
import Router from 'koa-router';

const app = new Koa();
const router = new Router();
app.context.pubnub = subscriptionsSetup(PUBNUB_CREDENTIALS);

router.get('/', ctx => (ctx.body = 'Hello World'));

app.use(morgan('tiny'));
app.use(cors());
app.use(dataloadersMiddleware);
app.use(authenticatedMiddleware);
app.use(router.routes()).use(router.allowedMethods());

router.all(
  '/playground',
  koaPlayground({
    endpoint: '/graphql',
    subscriptionEndpoint: `ws://localhost:${GRAPHQL_PORT}/subscriptions`,
  }),
);

router.all(
  '/graphql',
  graphqlHTTP({
    schema,
    graphiql: false,
  }),
);

export default app;
