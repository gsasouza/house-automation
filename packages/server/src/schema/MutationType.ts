import { GraphQLObjectType } from 'graphql';

import UserMutations from '../modules/user/mutations';

export default new GraphQLObjectType({
  name: 'Mutation',
  fields: () => ({
    ...UserMutations,
  })
});
