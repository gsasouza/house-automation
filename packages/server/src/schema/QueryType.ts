import { GraphQLObjectType, GraphQLString } from 'graphql';
import { connectionArgs } from 'graphql-relay';
import { NodeField } from '../interface/NodeInterface';

import UserConnection from '../modules/user/UserConnection';
import * as UserLoader from '../modules/user/UserLoader';

export default new GraphQLObjectType({
  name: 'Query',
  description: 'The root of all... queries',
  fields: () => ({
    node: NodeField,
    users: {
      type: UserConnection.connectionType,
      args: {
        ...connectionArgs,
        search: {
          type: GraphQLString,
        },
      },
      resolve: (_, args, context) => UserLoader.loadUsers(context, args),
    },
  })
});
