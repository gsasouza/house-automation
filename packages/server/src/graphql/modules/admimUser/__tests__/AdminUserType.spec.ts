import {
  clearDbAndRestartCounters,
  connectMongoose,
  createAdminUser,
  disconnectMongoose,
  getContext,
} from '../../../../../test/helpers';
import { toGlobalId } from 'graphql-relay';
import { graphql } from 'graphql';
import schema from '../../../schema';

beforeAll(connectMongoose);

beforeEach(clearDbAndRestartCounters);

afterAll(disconnectMongoose);

describe('Server: AdminUserType', () => {
  it('Should query an user', async () => {
    const adminUser = await createAdminUser();
    // language=GraphQL
    const query = `
      query Q($id: ID!) {
        adminUser: node(id: $id) {
          id
          ... on AdminUser {
            id
            username
            name
            email
          }
        }
      }
    `;

    const variables = {
      id: toGlobalId('AdminUser', adminUser._id),
    };
    const rootValue = {};
    const context = await getContext({ user: true });
    const result = await graphql(schema, query, rootValue, context, variables);

    expect(result.data).toMatchInlineSnapshot(
      {
        adminUser: {
          id: expect.any(String),
        },
      },
      `
      Object {
        "adminUser": Object {
          "email": "admin_user#1",
          "id": Any<String>,
          "name": "AdminUser#1",
          "username": "admin_user#1",
        },
      }
    `,
    );
  });

  it('Should query a list of users', async () => {
    await createAdminUser();
    await createAdminUser();
    await createAdminUser();
    await createAdminUser();

    // language=GraphQL
    const query = `
      query Q {
        adminUsers (first: 10) {
          edges {
            node {
              name
            }
          }
        }
      }
    `;

    const context = await getContext({ user: true });
    const result = await graphql(schema, query, {}, context, {});

    expect(result.data).toMatchSnapshot();
  });

  it('Should search a list of users', async () => {
    await createAdminUser();
    await createAdminUser();
    await createAdminUser({ name: 'search 1' });
    await createAdminUser({ name: 'search 2' });

    // language=GraphQL
    const query = `
      query Q ($search: String!) {
        adminUsers (search: $search) {
          edges {
            node {
              name
            }
          }
        }
      }
    `;

    const variables = { search: 'search' };
    const context = await getContext({ user: true });
    const result = await graphql(schema, query, {}, context, variables);

    expect(result.data).toMatchSnapshot();
  });
});
