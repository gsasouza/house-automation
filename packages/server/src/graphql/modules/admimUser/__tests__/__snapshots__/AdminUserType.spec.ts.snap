// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server: AdminUserType Should query a list of users 1`] = `
Object {
  "adminUsers": Object {
    "edges": Array [
      Object {
        "node": Object {
          "name": "AdminUser#3",
        },
      },
      Object {
        "node": Object {
          "name": "AdminUser#2",
        },
      },
      Object {
        "node": Object {
          "name": "AdminUser#1",
        },
      },
      Object {
        "node": Object {
          "name": "AdminUser#1",
        },
      },
    ],
  },
}
`;

exports[`Server: AdminUserType Should search a list of users 1`] = `
Object {
  "adminUsers": Object {
    "edges": Array [
      Object {
        "node": Object {
          "name": "search 2",
        },
      },
      Object {
        "node": Object {
          "name": "search 1",
        },
      },
    ],
  },
}
`;
